window_background/labels.js

```
@@ -790,19 +792,22 @@ function getDraftSet(eventName) {
 }
 
 function getDraftData(id) {
-  const draftData = pd.draft(id) || {
+  const draftData = playerData.draft(id) || {
     ...createDraft(),
     id,
     draftId: id,
-    date: new Date(),
-    owner: pd.name
+    date: undefined, // previously new Date()
```

Wouldn't this cause problems if the user has not set a custom timestamp format and the parser is unable to parse it? what is the result of that happening in case the user has played a couple draft / matches?

---

This is not the correct place to set a default date regardless. The correct place to set a default date is the location where date parsing fails.

In `onLabelInDraftDraftStatus` we set the date. If parsing fails because of the format then it will default to the current time (because it's using `parseWotcTimeFallback`).

This also applies to other things. The default values for data should try to avoid masking bugs so using reasonable values is a bad idea. If a fallback value is required for when data is missing then it should be done as late as possible.

Now that I think about it, it might even be a good idea to completely remove the fallback `new Date()` the date parsing and instead add a new field to all data called `parseDate`. This field can then always be set to `new Date()` regardless of what the date of the event is.Then when displaying the date of an event in the UI you check `event.date` and if it's missing you use `event.parseDate`. In this way you completely avoid the issue of accidentally adding invalid dates. `event.date` will *always* be a parsed log timestamp and `parseDate` will always be when the data was parsed.

---

You are right about the events though, if we don't get an `onLabelInDraftDraftStatus` event then it could cause problems. Hmm, I suppose we can just pass the date into `getDraftData` and move the `if (!data.date) {` code into that function.